lms/djangoapps/discussion/views.py

```diff
@@ -307,11 +307,36 @@ def forum_form_discussion(request, course_key):
             'corrected_text': query_params['corrected_text'],
         })
     else:
+        discussions_mfe_enabled_for_everyone = ENABLE_DISCUSSIONS_MFE_FOR_EVERYONE.is_enabled(course_key)
+        privileged_user = is_privileged_user(course_key, request.user)
+        if discussions_mfe_enabled_for_everyone and (not privileged_user):
+            discussion_experience = request.GET.get('discussions_experience', None)
+            banner_enabled = ENABLE_DISCUSSIONS_MFE_BANNER.is_enabled()
+            redirect_to_mfe = (discussion_experience is None) or (not banner_enabled)
+            if redirect_to_mfe and discussion_experience != "new":
```

Comment:
Instead of using the negation conditions we can simplify this. If banner_enabled and discussions_experience == "legacy" then redirect